manifests/exception.pp

```diff
@@ -82,8 +82,8 @@
   Enum['allow', 'block'] $action = 'allow',
   Boolean $enabled = true,
   Optional[Enum['TCP', 'UDP', 'ICMPv4', 'ICMPv6']] $protocol = undef,
-  Optional[Variant[Stdlib::Port, Enum['any']]] $local_port = undef,
-  Optional[Variant[Stdlib::Port, Enum['any']]] $remote_port = undef,
+  Optional[Variant[Stdlib::Port, Enum['any'], Pattern[/\A[1-9]{1}\Z|[1-9]{1}[0-9,-]*[0-9]{1}\Z/]]] $local_port = undef,
```

Comment:
It look like it's worth creating a [type alias](https://puppet.com/docs/puppet/6.18/lang_type_aliases.html) for this.  eg `Windows_firewall::Port`.

You can even add a test for it. https://rspec-puppet.com/documentation/type_aliases/

Reply:
Yeah, that's a good point @alexjfisher, it's pretty ugly. Working on it.